api/python/t4/search_util.py

```diff
@@ -74,10 +74,13 @@ def search(query, search_endpoint, limit, aws_region='us-east-1'):
     """
     es_client = _create_es(search_endpoint, aws_region)
 
-    payload = {'query': {'query_string': {
-        'default_field': 'content',
-        'query': query,
-    }}}
+    if isinstance(query, dict):
+        payload = query
+    else:
```

Comment:
shouldn't we check if `query` is a string at this point? if acceptable inputs are `dict` or `string` that should also land in the docstrings. we should do the most Pythonic thing. i.e. is a single argument that can assume multiple types Pythonic, or do we want to introduce, e.g. `query_dict=` as a fresh kwarg? @ResidentMario

Reply:
```
if isinstance(query, dict):
    payload = query
elif isinstance(query, str):
    payload = {'query': {'query_string': {
        'default_field': 'content',
        'query': query,
    }}}
else:
    raise ValueError
```

Reply:
Checking types is fine. Having multiple types of input is fine, we don't actually expose the `dict`-based API to the user anyway, we only talk about the string-based API in docs so w/e. Assuming pass-through values are actually strings is unsafe and leads to poor error messages.

Reply:
I took your advice to make sure the 'neither dict nor str' case raises. I agree that keeping this an undocumented feature is probably correct for now